_plugins/validate-releases.rb

```diff
@@ -0,0 +1,24 @@
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+STRING_KEYS = ['latest', 'releaseCycle']
+
+failure = false
+
+def process_file(markdown_file)
+  hash = YAML.load_file(markdown_file)
+  hash['releases'].each do |r|
+    STRING_KEYS.each do |k|
+      if r.key? k
+        if r[k].class != String
+          puts "#{markdown_file} has non-string value #{r[k]} in #{k}. Please wrap it in quotes"
+          failure = true
+        end
+      end
+    end
+  end
+end
+
+Dir['tools/*.md'].each { |file| process_file(file) }
+exit(1) if failure
```

Comment:
Testing locally with unfixed versions, it's printing the error lots of times, but not failing because `failure` here is still false.

Should `failure` be returned from `process_file`, or set as a global variable?

Reply:
Thanks a lot, fixed it by moving most of it into the function instead.

Reply:
Thanks, confirmed fails as expected.